CommonLibSF/CMakePresets.json

```diff
@@ -0,0 +1,52 @@
+{
+	"version": 3,
+	"cmakeMinimumRequired": {
+		"major": 3,
+		"minor": 21,
+		"patch": 0
+	},
+	"configurePresets": [
+		{
+			"name": "vcpkg",
+			"hidden": true,
+			"cacheVariables": {
+				"CMAKE_TOOLCHAIN_FILE": {
+					"type": "STRING",
+					"value": "$env{VCPKG_ROOT}/scripts/buildsystems/vcpkg.cmake"
+				},
+				"VCPKG_TARGET_TRIPLET": "x64-windows-static-md"
+			}
+		},
+		{
+			"name": "win64",
+			"hidden": true,
+			"architecture": "x64",
+			"cacheVariables": {
+				"CMAKE_MSVC_RUNTIME_LIBRARY": "MultiThreaded$<$<CONFIG:Debug>:Debug>DLL"
+			}
+		},
+		{
+			"name": "msvc",
+			"hidden": true,
+			"cacheVariables": {
+				"CMAKE_CXX_FLAGS": "/EHsc /MP /W4 /WX $penv{CXXFLAGS}"
+			},
+			"generator": "Visual Studio 17 2022",
```

Comment:
I suggest moving the mainline to Ninja for build performance at this point.

Reply:
I'm ok with deferring this to a later PR.

Reply:
I'm ok with this deferring this too, just want to keep it on the table. I'm going to mark this resolved.

Reply:
I've added both, tested briefly with LLVM 16.0.6 (latest stable) and MSVC 143
677e902add7463f8de6017263ee41819c98aad26